File: api/tts.py
from gtts import gTTS
from io import BytesIO

def handler(request):
    try:
        data = request.get_json() or {}
        text = data.get('text', '你好')
        lang = 'zh'  # demo voice
        tts = gTTS(text=text, lang=lang)
        buf = BytesIO()
        tts.write_to_fp(buf)
        audio = buf.getvalue()
        return (audio, 200, {
            "Content-Type": "audio/mpeg",
            "Access-Control-Allow-Origin": "*",
            "Cache-Control": "no-store"
        })
    except Exception as e:
        return ({ "error": str(e) }, 500, { "Content-Type": "application/json" })

def app(request):
    return handler(request)

File: api/tts.js
export const config = {
  runtime: 'nodejs18.x',
};

export default async function handler(req, res) {
  const origin = req.headers.origin || '*';
  res.setHeader('Access-Control-Allow-Origin', origin);
  res.setHeader('Vary', 'Origin');
  res.setHeader('Access-Control-Allow-Methods', 'POST,GET,OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
  if (req.method === 'OPTIONS') return res.status(204).end();

  // 这里只是占位返回，方便你后续接入真正 TTS
  res.status(501).json({ ok: false, message: 'TTS endpoint not implemented yet.' });
}
